iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/CatalogUtils.java

```diff
@@ -0,0 +1,227 @@
+    Properties properties = new Properties();
+    properties.putAll(toIcebergProperties(hmsTable.getParameters()));
+
+    if (properties.get(LOCATION) == null && hmsTable.getSd() != null &&
+        hmsTable.getSd().getLocation() != null) {
+      properties.put(LOCATION, hmsTable.getSd().getLocation());
+    }
+
+    if (properties.get(NAME) == null) {
+      properties.put(NAME, TableIdentifier.of(hmsTable.getDbName(),
+          hmsTable.getTableName()).toString());
+    }
+
+    SerDeInfo serdeInfo = hmsTable.getSd().getSerdeInfo();
+    if (serdeInfo != null) {
+      properties.putAll(toIcebergProperties(serdeInfo.getParameters()));
+    }
+
+    // Remove HMS table parameters we don't want to propagate to Iceberg
+    PROPERTIES_TO_REMOVE.forEach(properties::remove);
+
+    return properties;
+  }
+
+  private static Properties toIcebergProperties(Map<String, String> parameters) {
+    Properties properties = new Properties();
+    parameters.entrySet().stream()
+        .filter(e -> e.getKey() != null && e.getValue() != null)
+        .forEach(e -> {
+          String icebergKey = HMSTablePropertyHelper.translateToIcebergProp(e.getKey());
+          properties.put(icebergKey, e.getValue());
+        });
+    return properties;
+  }
+
+  /**
+   * Collect all the catalog specific configuration from the global hive configuration.
+   * @param conf a Hadoop configuration
+   * @param catalogName name of the catalog
+   * @return complete map of catalog properties
+   */
+  public static Map<String, String> getCatalogProperties(Configuration conf, String catalogName) {
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    String keyPrefix = CATALOG_CONFIG_PREFIX + catalogName;
+    conf.forEach(config -> {
+      if (config.getKey().startsWith(CatalogUtils.CATALOG_DEFAULT_CONFIG_PREFIX)) {
+        catalogProperties.putIfAbsent(
+            config.getKey().substring(CatalogUtils.CATALOG_DEFAULT_CONFIG_PREFIX.length()),
+            config.getValue());
+      } else if (config.getKey().startsWith(keyPrefix)) {
+        catalogProperties.put(
+            config.getKey().substring(keyPrefix.length() + 1),
+            config.getValue());
+      }
+    });
+
+    return catalogProperties;
+  }
+
+  public static String getCatalogName(Configuration conf) {
+    return MetastoreConf.getVar(conf, MetastoreConf.ConfVars.CATALOG_DEFAULT);
+  }
+
+  public static String getCatalogType(Configuration conf) {
+    return getCatalogType(conf, CatalogUtils.getCatalogName(conf));
+  }
+
+  public static String getCatalogType(Configuration conf, Properties catalogProperties) {
+    String catalogName = catalogProperties.getProperty(
+        CatalogUtils.CATALOG_NAME,
+        MetastoreConf.getVar(conf, MetastoreConf.ConfVars.CATALOG_DEFAULT));
+    return getCatalogType(conf, catalogName);
+  }
+
+  public static boolean isCustomHadoopCatalogImpl(Configuration conf, String catalogName) {
```

Comment:
can't we extend the check in assertCatalogType directly?
...
String catalogImpl = getCatalogImpl(conf, catalogName);
return StringUtils.equals(expectedImpl, catalogImpl) || 
  Class.forName(expectedImpl).isAssignableFrom(Class.forName(catalogImpl));
````

Reply:
Done